go-controller/pkg/ovn/master.go

```diff
@@ -228,6 +241,159 @@ func (oc *Controller) syncNodeManagementPort(node *kapi.Node) error {
 	return nil
 }
 
+func parseGatewayIfaceID(node *kapi.Node) (string, error) {
+	ifaceID, ok := node.Annotations[OvnNodeGatewayIfaceID]
+	if !ok {
+		logrus.Errorf("gateway ifaceID annotation not found for node %q ", node.Name)
+		return "", nil
+	}
+
+	return ifaceID, nil
+}
+
+func parseGatewayMacAddress(node *kapi.Node) (string, error) {
+	gatewayMacAddress, ok := node.Annotations[OvnNodeGatewayMacAddress]
+	if !ok {
+		logrus.Errorf("gatewayMacAddress annotation not found for node %q ", node.Name)
+		return "", nil
+	}
+
+	_, err := net.ParseMAC(gatewayMacAddress)
+	if err != nil {
+		return "", fmt.Errorf("Error %v in parsing node gateway macAddress %v", err, gatewayMacAddress)
+	}
+
+	return gatewayMacAddress, nil
+}
+
+func parseGatewayLogicalNetwork(node *kapi.Node) (string, string, error) {
+	ipAddress, ok := node.Annotations[OvnNodeGatewayIP]
+	if !ok {
+		logrus.Errorf("gatewayIpAddress annotation not found for node %q ", node.Name)
+		return "", "", nil
+	}
+
+	gwNextHop, ok := node.Annotations[OvnNodeGatewayNextHop]
+	if !ok {
+		logrus.Errorf("gatewayNextHop annotation not found for node %q ", node.Name)
+		return "", "", nil
+	}
+
+	return ipAddress, gwNextHop, nil
+}
+
+func (oc *Controller) syncGatewayLogicalNetwork(node *kapi.Node, mode string) error {
+
+	if mode == string(config.GatewayModeDisabled) {
+		return nil
+	}
+
```

Comment:
L287-289: I don't think this check is needed since the caller has already ensured that `mode` is not `disabled.